ecosystem/sep-0008.md

```diff
@@ -126,6 +126,8 @@ Name | Data Type | Description
 status|string|`"action_required"`
 message|string|A human readable string containing information regarding the action required.
 action_url|string|A URL that allows the user to complete the actions required to have the transaction approved.
```

Comment:
Today it's assumed the client will open the URL in a browser, but providing these options might need to be POSTed to keep sensitive information out of the URL query params. We probably need to specify what request format this URL should be in the case actions params are provided.

Reply:
Yeah, i thought that the client would add those to the query params, as it's currently exactly like that (e.g. as you can see from our sample we are already adding the public key, taken from the tx.

Reply:
I don't see a problem with adding the public key to the URL but the name and email could be leaked in other places by including it there. URLs are not always secure as they get easily propagated to logs, ad tracking, analytics systems, etc, so it would be ideal to keep these details in a POST either by submitting them in the request body as a POST separately and getting back a URL with an ID referencing that data, or by using a POST at this point although that might make it harder to implement in some cases such as mobile.

SEP-12 would help provide that parallel delivery of data but it sounds like that would be significantly different to what you're doing today. I think we should probably explore that. Return of a list of data fields needed could be delivered separately as the custom, and then the action URL is loaded for the user. Prior to SEP-12 the flow would probably need SEP-10 to authenticate as the account first which could be seamless.

Reply:
Those are valid points, especially with the POST request.

SEP-12, as I understand it, has only `NEEDS_INFO`, which isn't the case here. We can work without E-Mail by collecting it for ourself (not all clients may have that info at hand), so it's optional.

Reply:
Got it, so what you're saying is if you don't provide this information then it would be okay, the interactive browser page would then ask for it so it isn't the same thing because this is prefilling not requiring data, and if all the data is prefilled it can be skipped, and if there are fields still required the page opened can collect it. Is that accurate?

Reply:
@shredding @dmgmyza and I discussed this offline and an alternative approach where:

Add a new optional response field, `action_method`. The field has three valid states:
- Not provided, interpreted the same as if it is given the value `GET`
- `GET` – the action URL expects to receive a GET request, any fields requested may be passed as query parameters
- `POST` – the action URL expects to receive a POST request, any fields requested may be passed url-form-encoded or as JSON

Action URLs providing a `POST` method should response to the POST with a HTTP response with status code 301 and a `Location` header that is a URL where the user can complete the required actions with all of the POST data pre-filled or already accepted.

This approach supports products entirely in a web browser because the browser will follow the 301 redirect and will prevent odd UI where a user for some reason refreshes the page, they won't see warnings about they are refreshing a page that is accepting post data.

This approach supports products that are mobile apps loading the URL in a system browser because the app can make the POST request internally, read the location out of the response and provide it to a browser to load.

This approach is additive, any existing implementation would see no change. Any existing implementations that are providing optional fields/params as query parameters would be compatible. Any new implementations that plan to provide personal data or sensitive data could choose to use the POST method.

This approach does put some burden on the client to support different approaches though which is not great. We could encourage the POST approach for new implementations.